backend/internal/service/handler/spotistics/clientcreds.go

```diff
@@ -0,0 +1,51 @@
+package spotistics
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/gofiber/fiber/v2"
+	spotifyauth "github.com/zmb3/spotify/v2/auth"
+
+	"github.com/zmb3/spotify/v2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type SpotifyCredentials struct {
+	ClientID     string
+	ClientSecret string
+}
+
+type SpotifyKey struct{}
+
+func (creds *SpotifyCredentials) getClient() spotify.Client {
+	ctx := context.Background()
+	config := &clientcredentials.Config{
+		ClientID:     creds.ClientID,
+		ClientSecret: creds.ClientSecret,
+		TokenURL:     spotifyauth.TokenURL,
+	}
+
+	token, err := config.Token(ctx)
+	if err != nil {
```

Comment:
i'd argue this shouldn't be a fatal error, we can bubble this up by returning the error. it will then be caught by the central error handler